File: trading-bot/backend/test_login.js
const axios = require("axios");

async function testLogin() {
  try {
    const params = new URLSearchParams();
    params.append("username", "Khalil@gmail.com");
    params.append("password", "your_password_here"); // Assuming you have a way to know or test a known user

    // Let's try with test@example.com which I saw in the DB
    const params2 = new URLSearchParams();
    params2.append("username", "test@example.com");
    params2.append("password", "testpassword"); // Assuming default or common test pass

    const res = await axios.post("http://localhost:8000/token", params2, {
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
    });
    console.log("Login Status:", res.status);
    console.log("Token Received:", !!res.data.access_token);
  } catch (err) {
    console.error("Login Failed:", err.response?.data || err.message);
  }
}

testLogin();

File: trading-bot/backend/test_login.py
import requests

def test_login():
    url = "http://localhost:8000/token"
    data = {
        "username": "Khalil@gmail.com",
        "password": "testpassword" # Assuming a test password for the user in the DB
    }
    # Or test@example.com
    data_test = {
        "username": "test@example.com",
        "password": "testpassword"
    }

    try:
        # FastAPI OAuth2PasswordRequestForm expects form-data (either multipart or urlencoded)
        # requests.post with 'data' kwarg uses application/x-www-form-urlencoded
        response = requests.post(url, data=data_test)
        print(f"Status: {response.status_code}")
        if response.status_code == 200:
            print("Login successful!")
            print(f"Token: {response.json().get('access_token')[:10]}...")
        else:
            print(f"Login failed: {response.text}")
    except Exception as e:
        print(f"Request error: {e}")

if __name__ == "__main__":
    test_login()
